### PR TITLE
add codecov for notifications-es

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*

*Description of changes:*

 By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by chloe-zh** [comment]
Add a coverage badge in the readme page? Ref: https://github.com/opendistro-for-elasticsearch/sql/blame/develop/README.md#L3

**Review by akbhatta — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Comment by zhongnansu** [comment]
> Add a coverage badge in the readme page? Ref: https://github.com/opendistro-for-elasticsearch/sql/blame/develop/README.md#L3

good point, will do

File: notifications/build-tools/esplugin-coverage.gradle [hunk 1]
```diff
@@ -58,7 +58,7 @@ integTest {
 
 jacocoTestReport {
     dependsOn integTest, test
-    executionData dummyTest.jacoco.destinationFile, dummyIntegTest.jacoco.destinationFile
```
Comment (akbhatta) [inline notifications/build-tools/esplugin-coverage.gradle:61]: why integ test is excluded?

Comment (zhongnansu) [reply notifications/build-tools/esplugin-coverage.gradle:61]: the existing trick of dumping IT test result to jacoco doesn't seem working fully. Also, elasticsearch official doc says coverage is not doable https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc#test-coverage-analysis
So currently only UT is dumped into jacoco report. I added a comment in the code for us to look back

**Pull request merged**